zeebe/gateway-grpc/src/main/java/io/camunda/zeebe/gateway/Gateway.java

```diff
@@ -378,7 +378,9 @@ private ServerServiceDefinition applyInterceptors(final BindableService service)
     Collections.reverse(interceptors);
     interceptors.add(new ContextInjectingInterceptor(queryApi));
     interceptors.add(MONITORING_SERVER_INTERCEPTOR);
-    interceptors.add(new AuthenticationInterceptor(userServices));
+    if (securityConfiguration.getAuthorizations().isEnabled()) {
```

Comment:
❌ I don't think this is right. Authorization and authentication are two different things. One could disable authorization checks but still require authentication as mentioned [here](https://github.com/camunda/camunda/pull/25762#discussion_r1880405062). I also wasn't aware this was a requirement before 😅

Reply:
Do we have a flag to enable/disable authentication ?

Reply:
I'm not sure. @houssain-barouni do you know if this flag exists yet?

Reply:
I think now it is just controlled with Spring profiles
I see that this is being discussed here https://camunda.slack.com/archives/C06UYJMMETZ/p1734355575029589?thread_ts=1733170087.268599&cid=C06UYJMMETZ

Reply:
Should I re-add the auth mode then @remcowesterhoud ? Instead of having Identity and None, we can add OIDC and BASIC

Reply:
@nicpuppa Let's wait and see what people say on the slack thread 😄

Reply:
Once https://github.com/camunda/camunda/pull/26243 is merged we can change the check to is auth mode not none